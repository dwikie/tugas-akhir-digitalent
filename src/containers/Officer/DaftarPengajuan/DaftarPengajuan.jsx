import React, { useCallback, useEffect, useMemo, useState } from "react";
import { FilePdfOutlined } from "@ant-design/icons";
import { Button, Col, Input, Row } from "antd";
import Title from "../../../components/Title";
import useBreakpoint from "antd/lib/grid/hooks/useBreakpoint";
import { useHistory, useRouteMatch } from "react-router";
import DaftarPengajuanTable from "../../../components/DaftarPengajuanTable";
import { getAll } from "../../../services/pengajuan-service";
import Columns from "./TableColumns";
import DaftarPengajuanPagination from "../../../components/DaftarPengajuanPagination/DaftarPengajuanPagination";

export default function DaftarPengajuan() {
  const { url } = useRouteMatch();
  const { push } = useHistory();
  // const { search } = useLocation();

  // const initialPage = useRef(new URLSearchParams(search).get("page"));

  const [tableData, setTableData] = useState([]);
  const [talbeError, setTableError] = useState("");
  const [isTableLoading, setIsTableLoading] = useState(true);
  // const [page, setPage] = useState(() =>
  //   isNaN(initialPage.current) ? 1 : parseInt(initialPage.current),
  // );
  const service = useMemo(() => getAll(1, 10), []);

  const breakPoint = useBreakpoint();

  const getPengajuan = useCallback(async () => {
    setIsTableLoading(true);
    try {
      const { data } = await service.start();
      setTableData(data);
      setIsTableLoading(false);
    } catch (error) {
      setTableError(error.message);
      setIsTableLoading(false);
    }
  }, [service]);

  useEffect(() => {
    getPengajuan();
    return () => {
      service.cancel();
      setTableData([]);
    };
  }, [getPengajuan, service]);

  function onTableRowClick(id_pengajuan) {
    push(`${url}/${id_pengajuan}`);
  }

  return (
    <Row gutter={[0, 16]} style={{ flexDirection: "column" }}>
      <Title title="Daftar Pengajuan" />

      <section className="container" id="daftar-pengajuan-action">
        <Row
          justify={"space-between"}
          style={{ flexDirection: breakPoint.md ? "row" : "column-reverse" }}
          gutter={[0, 20]}
        >
          <Col sm={24} md={8}>
            <Input placeholder="Cari.." allowClear />
          </Col>
          <Col sm={24} md={8}>
            <Button
              style={{ width: "100%" }}
              icon={<FilePdfOutlined />}
              text="Download Daftar Pengajuan"
            >
              Download List Pengajuan
            </Button>
          </Col>
        </Row>
      </section>
      <section className="container" id="daftar-pengajuan-table">
        <Row style={{ flexDirection: "column" }} gutter={[0, 16]}>
          <DaftarPengajuanTable
            columns={Columns}
            onTableRowClick={onTableRowClick}
            data={tableData}
            error={talbeError}
            loading={isTableLoading}
          />
          <Row justify="end">
            <DaftarPengajuanPagination />
          </Row>
        </Row>
      </section>
    </Row>
  );
}
