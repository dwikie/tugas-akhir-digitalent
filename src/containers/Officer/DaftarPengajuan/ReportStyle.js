import { StyleSheet } from "@react-pdf/renderer";

export default StyleSheet.create({
  body: {
    padding: "8px 12px",
  },
  info: {
    fontSize: 8,
  },
  infoWrapper: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    marginBottom: 20,
    flexDirection: "row",
  },
  title: {
    fontSize: 18,
    textAlign: "center",
    textTransform: "uppercase",
    marginBottom: 16,
  },
  table: {
    display: "table",
    fontSize: 10,
    width: "auto",
    borderStyle: "solid",
    borderColor: "rgb(150 150 150 / 50%)",
    borderWidth: 1,
    borderRightWidth: 0,
    borderBottomWidth: 0,
  },
  tableHead: {
    flexDirection: "row",
  },
  tableRow: {
    position: "relative",
    backgroundColor: "#006cbb",
    color: "white",
    width: "47%",
    padding: 4,
    borderRight: "1 solid rgb(150 150 150 / 50%)",
    borderBottom: "1 solid rgb(150 150 150 / 50%)",
  },
  tableHeadCell: {
    margin: "auto",
    fontSize: 10,
  },
  tableBody: {
    flexDirection: "row",
  },
  tableBodyRow: {
    position: "relative",
    flexDirection: "column",
    backgroundColor: "white",
    color: "black",
    width: "47%",
    padding: 4,
    borderRight: "1 solid rgb(150 150 150 / 50%)",
    borderBottom: "1 solid rgb(150 150 150 / 50%)",
  },
  tableBodyCell: {
    fontSize: 10,
  },
  tableDataRow: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginBottom: 2,
  },
  tableDataCol: {
    flexDirection: "column",
    width: "50%",
    flexWrap: "wrap",
  },
});
