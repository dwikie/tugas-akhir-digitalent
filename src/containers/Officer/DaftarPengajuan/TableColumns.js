import { StatusPengajuan } from "../../../utils";

const Columns = [
  {
    title: "No.",
    dataIndex: "no",
    key: "no",
    render: (text, row, index) => index + 1,
  },
  {
    title: "Tanggal Pengajuan",
    dataIndex: "tanggalPengajuan",
    key: "tanggalPengajuan",
    render: () => "Senin, 23 Agustus",
  },
  {
    title: "Nama",
    dataIndex: "nama_lengkap",
    key: "nama",
  },
  {
    title: "Status",
    dataIndex: "status",
    key: "status",
    render: (value) => {
      return StatusPengajuan(value).detail;
    },
  },
  {
    title: "Rekomendasi",
    dataIndex: "rekomendasi",
    key: "rekomendasi",
    render: () => "Iya",
  },
];

export default Columns;
