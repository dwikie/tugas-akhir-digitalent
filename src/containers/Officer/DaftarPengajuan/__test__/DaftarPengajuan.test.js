import React, { useState } from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, waitFor, cleanup, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import DaftarPengajuan from "../DaftarPengajuan";

afterEach(cleanup);

const LazyDaftarPengajuan = React.lazy(() => import("../DaftarPengajuan"));

// function Main() {
//   return (
//     <div>
//       <div>Lazy Component here</div>
//       <LazyDaftarPengajuan />
//     </div>
//   );
// }

describe("Render Daftar Pengajuan", () => {
  // beforeEach(() => {
  //   jest.setTimeout(10000);
  // });

  it("React.Suspense > Lazy Component > No Data", async () => {
    const { container } = render(
      <MemoryRouter
        initialEntries={[
          {
            push: jest.fn(),
            url: "/dashboard/pengajuan",
          },
        ]}
      >
        <React.Suspense fallback="Test loading">
          <LazyDaftarPengajuan />
        </React.Suspense>
        ,
      </MemoryRouter>,
    );

    // debug();
    // const element = await screen.findByText(/Lazy Component here/);
    // expect(element).toBeInTheDocument();
    const element = await waitFor(() => screen.findByText(/No Data/i));
    expect(element).not.toBeNull();
    expect(element).toMatchSnapshot();
  });

  // it("React.Suspense > Not Lazy Component", () => {
  //   const { debug } = render(
  //     <MemoryRouter
  //       initialEntries={[
  //         {
  //           push: jest.fn(),
  //           url: "/dashboard/pengajuan",
  //         },
  //       ]}
  //     >
  //       <React.Suspense fallback="Test Loading">
  //         <DaftarPengajuan />
  //       </React.Suspense>
  //     </MemoryRouter>,
  //   );

  //   // debug();
  // });
});

// describe("Render Daftar Pengajuan", () => {
//   it("Pusing awak kasi nama", () => {
//     // const myInitialState = "My Initial State";
//     // React.useState = jest.fn().mockReturnValue([myInitialState, {}]);
//     // const onTableRowClick = jest.fn((id) => console.log(id));
//     // console.log(fetch);
//     const onTableRowClick = jest.fn();
//     const { container } = render(
//       <MemoryRouter
//         initialEntries={[
//           {
//             push: jest.fn(),
//             url: "/dashboard/pengajuan",
//           },
//         ]}
//       >
//         <DaftarPengajuan data={fetch} />
//       </MemoryRouter>,
//     );
// expect(onTableRowClick).toHaveBeenCalledTimes(1);
// const tableRow = container.querySelectorAll(".ant-table-row");
// console.log(tableRow);
// fireEvent.click(tableRow);
// expect(tableRow).not.tobeNull();
// expect(wrapper).toBeTruthy();
// const element = screen.getByText("Asep Sunandar");
// expect(element).toBeTruthy();
// expect(onTableRowClick).toHaveBeenCalledTimes(1);
//   });
// });
