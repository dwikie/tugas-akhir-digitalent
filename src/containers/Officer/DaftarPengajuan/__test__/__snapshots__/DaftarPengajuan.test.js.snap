// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Daftar Pengajuan React.Suspense > Lazy Component > No Data 1`] = `
<div
  class="ant-empty-description"
>
  No Data
</div>
`;

exports[`Render Daftar Pengajuan React.Suspense > Lazy Component 1`] = `
<h5
  class="ant-typography"
  style="text-align: center;"
>
  Daftar Pengajuan
</h5>
`;
